test_suite/tester.py

```diff
@@ -99,7 +100,106 @@ def default_tempesta_factory(tempesta):
 register_backend("docker", docker_srv_factory)
 
 
-class TempestaTest(unittest.TestCase):
+@dataclasses.dataclass
+class TempestaLoggers:
+    dmesg: dmesg.DmesgFinder
+    get_tempesta: typing.Callable
+
+    @property
+    def clickhouse(self) -> clickhouse.ClickHouseFinder:
+        return self.get_tempesta().clickhouse
+
+
+class WaitUntilAsserts(unittest.TestCase):
+    def assertWaitUntilEqual(
+        self,
+        func: typing.Callable,
+        second,
+        msg: str = None,
+        timeout: int = 5,
+        poll_freq: float = 0.1,
+    ):
+        success = util.wait_until(
+            wait_cond=lambda: func() != second, timeout=timeout, poll_freq=poll_freq
+        )
+
+        if success:
+            return None
+
+        self.fail(self._formatMessage(msg, f"Not equals even after {timeout} seconds"))
+
+    def assertWaitUntilNotEqual(
+        self,
+        func: typing.Callable,
+        second,
+        msg: str = None,
+        timeout: int = 5,
+        poll_freq: float = 0.1,
+    ):
+        success = util.wait_until(
+            wait_cond=lambda: func() == second, timeout=timeout, poll_freq=poll_freq
+        )
+
+        if success:
+            return None
+
+        self.fail(self._formatMessage(msg, f"Still equals even after {timeout} seconds"))
+
+    def assertWaitUntilIsNotNone(
+        self, func: typing.Callable, msg: str = None, timeout: int = 5, poll_freq: float = 0.1
+    ):
+        success = util.wait_until(
+            wait_cond=lambda: func() is None, timeout=timeout, poll_freq=poll_freq
+        )
+
+        if success:
+            return None
+
+        self.fail(self._formatMessage(msg, f"Is None event after {timeout} seconds"))
+
+    def assertWaitUntilCountEqual(
```

Comment:
This function is duplicate for `assertWaitUntilEqual`. I think we can remove it.

Reply:
not at all, assertWaitUntilCountEqual applies the len function to lambda